lib/rails_translation_manager/locale_checker/incompatible_plurals.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+class IncompatiblePlurals < BaseChecker
+  include LocaleCheckerHelper
+
+  def report
+    format_incompatible_plurals if incompatible_plurals.present?
+  end
+
+  private
+
+  def format_incompatible_plurals
+    <<~OUTPUT.chomp
+      \e[31m[ERROR]\e[0m Incompatible plural forms, for:
+
+      #{incompatible_plurals.join("\n\n")}
+
+      #{non_plural_keys_message}
+    OUTPUT
+  end
+
+  def incompatible_plurals
+    @incompatible_plurals ||= grouped_plural_keys_for_locale.flat_map do |plural_keys_for_locale|
+      locale = plural_keys_for_locale[:locale]
+
+      if all_plural_forms[locale].blank?
+        missing_plural_form_message(locale)
+      else
+        missing_plural_keys(plural_keys_for_locale).compact
```

Comment:
It feels like calling `.compact` should be the responsibility of the `missing_plural_keys` method. I'd move that into there. It may also present an opportunity to split that method into two - one for identifying the missing plural keys, and one for formatting the error strings - as that method is quite fiddly to reason about at the moment. (Probably just because I've read the word 'plural' about a hundred times in the last 20 minutes!)

Reply:
Good point, have moved `compact` to said method! I've also had a bash at simplifying the code (clearer variables etc) which should make it a little bit more readable. I've left the 'formatting' part in [that method](https://github.com/alphagov/rails_translation_manager/pull/12/files#diff-667e6893a3236e587c2311d7b3a359c2d91bb229535bb1862b7b8245524f31b1R34), as otherwise you'd have to supply the simple string method 4 arguments, which in practice, didn't make things any clearer! 😄